test-data/divmod.yaml

```diff
@@ -0,0 +1,317 @@
+    r1 /= r2  ; this could divide by 0
+
+post:
+  - r1.type=number
+  - r2.type=number
+---
+test-case: zero divided by undefined value register
+
+pre: ["r1.type=number", "r1.value=0", "r2.type=number"]
+
+code:
+  <start>: |
+    r1 /= r2  ; this could divide by 0
+
+post:
+  - r1.type=number
+  - r1.value=0
+  - r2.type=number
+---
+test-case: non-zero modulo zero immediate
+
+pre: ["r1.type=number", "r1.value=6"]
+
+code:
+  <start>: |
+    r1 %= 0
+
+post: []
+
+messages:
+  - "0: Code is unreachable after 0"
+---
+test-case: zero modulo zero immediate
+
+pre: ["r1.type=number", "r1.value=0"]
+
+code:
+  <start>: |
+    r1 %= 0
+
+post: []
+
+messages:
+  - "0: Code is unreachable after 0"
+---
+test-case: non-zero modulo zero register
+
+pre: ["r1.type=number", "r1.value=6", "r2.type=number", "r2.value=0"]
+
+code:
+  <start>: |
+    r1 %= r2
+
+post: []
+
+messages:
+  - "0: Code is unreachable after 0"
+---
+test-case: zero modulo zero register
+
+pre: ["r1.type=number", "r1.value=0", "r2.type=number", "r2.value=0"]
+
+code:
+  <start>: |
+    r1 %= r2
+
+post: []
+
+messages:
+  - "0: Code is unreachable after 0"
+---
+test-case: non-zero modulo possibly zero register
+
+pre: ["r1.type=number", "r1.value=6", "r2.type=number", "r2.value=[-5, 5]"]
+
+code:
+  <start>: |
+    r1 %= r2 ; this could do modulo 0 so could set r1 = r2
+
+post:
+  - r1.type=number
+  - r1.value=[0, 4]
+  - r2.type=number
+  - r2.value=[-5, 5]
+---
+test-case: zero modulo possibly zero register
+
+pre: ["r1.type=number", "r1.value=0", "r2.type=number", "r2.value=[-5, 5]"]
+
+code:
+  <start>: |
+    r1 %= r2 ; this could do modulo 0 so could set r1 = r2
+
+post:
+  - r1.type=number
+  - r1.value=[0, 4]
+  - r2.type=number
+  - r2.value=[-5, 5]
+---
+test-case: non-zero modulo possibly zero register 2
+
+pre: ["r1.type=number", "r1.value=6", "r2.type=number", "r2.value=[-5, 0]"]
+
+code:
+  <start>: |
+    r1 %= r2 ; this could do modulo 0 so could set r1 = r2
+
+post:
+  - r1.type=number
+  - r1.value=[0, 4]
+  - r2.type=number
+  - r2.value=[-5, 0]
+---
+test-case: zero modulo possibly zero register 2
+
+pre: ["r1.type=number", "r1.value=0", "r2.type=number", "r2.value=[-5, 0]"]
+
+code:
+  <start>: |
+    r1 %= r2 ; this could do modulo 0 so could set r1 = r2
+
+post:
+  - r1.type=number
+  - r1.value=[0, 4]
+  - r2.type=number
+  - r2.value=[-5, 0]
+---
+test-case: non-zero modulo undefined value register
+
+pre: ["r1.type=number", "r1.value=6", "r2.type=number"]
+
+code:
+  <start>: |
+    r1 %= r2  ; this could be modulo 0
+
+post:
+  - r1.type=number
+  - r2.type=number
+---
+test-case: zero modulo undefined value register
+
+pre: ["r1.type=number", "r1.value=0", "r2.type=number"]
+
+code:
+  <start>: |
+    r1 %= r2  ; this could be modulo 0
+
+post:
+  - r1.type=number
+  - r2.type=number
+---
+test-case: negative modulo negative
+
+pre: ["r1.type=number", "r1.value=-13"]
+
+code:
+  <start>: |
+    r1 %= 3
+
+post:
+  - r1.type=number
+  - r1.value=-1
```

Comment:
This (and the other results below) are not what they should be according to the specification draft, instead they work like `-(abs(a)%abs(b))`.

Reply:
You're right... according to https://en.wikipedia.org/wiki/Modulo_operation the draft should say trunc() not floor().  I will fix that in the draft.

Reply:
Corrected draft